### ISSUE TITLE
Task rules fails in 3.0

### ISSUE THREAD

**Issue by kicktipp** [node-gradle/gradle-node-plugin#145]
When I switch to version 3.0.0. I get an error.

```
A problem occurred evaluating project ':kicktipp-frontend'.
> Could not find method npm_run_build() for arguments [build_5ed7z6e9l38swkhmnttelbpig$_run_closure3@5db36052] on project ':kicktipp-frontend' of type org.gradle.api.Project.
```
The problem part is here:

```
npm_run_build {
    inputs.files fileTree('src')
    inputs.files fileTree('public')
    outputs.dir "dist"
}

// See https://dzone.com/articles/integrating-java-and-npm-builds-using-gradle
task packageFrontend(type: Zip) {
    dependsOn npm_run_build
    ...
}
```

I fixed it by setting my replacing the npm_run_build configuration with
```
task npmRunBuild (type: NpmTask) {
    inputs.files fileTree('src')
    inputs.files fileTree('public')
    outputs.dir "dist"
    args = ['run', 'build']
}
```
and replacing dependsOn with 

```
task packageFrontend(type: Zip) {
    dependsOn npmRunBuild

```
This might be intended because of configuration caching. But it should be mentioned in breaking changes.

**Comment by ajurge** [comment]
I also get the same error for all our npm tasks that worked with v2.2.4: `` `Could not find method npm_install() for arguments [build_162xn1b2o79uwd4haivx1dbko$_run_closure8@78ff85a] on project ':webapp' of type org.gradle.api.Project.`

**Comment by deepy** [comment]
It's unintended, this isn't something that was supposed to break, but when it comes to `npm_install` I highly recommend using the built-in `npmInstall` instead https://github.com/node-gradle/gradle-node-plugin/blob/master/docs/usage.md#executing-npm-tasks

This is the code that sets up the task rule responsible for `npm_foo` https://github.com/node-gradle/gradle-node-plugin/blob/master/src/main/kotlin/com/github/gradle/node/NodePlugin.kt#L58-L71
If I were to guess I'd guess it's `task.register()` vs `task.create()` (eager vs lazy)

**Comment by deepy** [comment]
Yeah, it's because of switching from `task.create()` to `task.register()`, I'll add some tests and then fix this, though this change will make it to next major

**Comment by deepy** [comment]
This is fixed in 3.0.1
